### ISSUE TITLE
adding extremely large number of days fails (related to leap years)

### ISSUE THREAD

**Issue by autarch** [houseabsolute/DateTime.pm#34]
Migrated from [rt.cpan.org #97046](https://rt.cpan.org/Ticket/Display.html?id=97046) (status was 'open')

Requestors:
* kmx@cpan.org (@kmx)

From kmx@cpan.org (@kmx) on 2014-07-08 08:11:27:

Here is a failing code:

```perl
use DateTime;

my $DAY_ONE = DateTime->new(year => 0, month => 1, day => 1);

warn $DAY_ONE->clone->add(days => 268526343)->ymd, "\n";        # 735200-02-27
warn $DAY_ONE->clone->add(days => 268526344)->ymd, "\n";        # 735200-02-28
warn eval { $DAY_ONE->clone->add(days => 268526345)->ymd };     # FAILS (should be 735200-02-29)
warn $DAY_ONE->clone->add(days => 268526346)->ymd, "\n";        # 735200-03-01
warn $DAY_ONE->clone->add(days => 268526347)->ymd, "\n";        # 735200-03-02
```

the failure is: Invalid day of month (day = 30 - month = 2 - year = 735200)

I am using: perl 5.18.2 (MSWin32 / x86) + DateTime 1.10

--
kmx

**Comment by autarch** [comment]
From drolsky@cpan.org (@autarch) on 2015-01-04 01:34:12:

There's a fix for this in the 1.15 trial release. If I see no new breakage on CPAN tester I'll do a proper release with this fix in the future.

That sounds, there is still some number that will be too large, it's just now system-dependent.

**Comment by autarch** [comment]
Looks like I released the fix for this as stable quite some time ago.
